tunnel/winipcfg/types.go

```diff
@@ -944,13 +948,28 @@ func (row *MibIPforwardRow2) Delete() error {
 }
 
 // get method returns all table rows as a Go slice.
-func (tab *mibIPforwardTable2) get() []MibIPforwardRow2 {
-	const maxCount = maxIndexCount8 / unsafe.Sizeof(MibIPforwardRow2{})
-	return (*[maxCount]MibIPforwardRow2)(unsafe.Pointer(&tab.table[0]))[:tab.numEntries]
+func (tab *mibIPforwardTable2) get() (s []MibIPforwardRow2) {
+	h := (*unsafeSlice)(unsafe.Pointer(&s))
+	h.Data = unsafe.Pointer(&tab.table[0])
+	h.setLen(int(tab.numEntries))
+	return
 }
 
 // free method frees the buffer allocated by the functions that return tables of network interfaces, addresses, and routes.
 // https://docs.microsoft.com/en-us/windows/desktop/api/netioapi/nf-netioapi-freemibtable
 func (tab *mibIPforwardTable2) free() {
 	freeMibTable(unsafe.Pointer(tab))
 }
+
+type unsafeSlice struct {
+	Data unsafe.Pointer
+	Len  int
+	Cap  int
+}
+
+func (h *unsafeSlice) setLen(n int) {
```

Comment:
If we can't get rid of unsafeSlice by moving some of this into sys as mentioned above, we could at least minimize the implementation by not adding methods and just assigning at initialization time and copying that rvalue result to the casted pointer.

Reply:
Sure, if you dislike the method I can remove it. I was on the fence anyway.